### DISCUSSION TITLE
Noir LSP often crashes with minor syntax errors

### DISCUSSION THREAD

**Discussion by critesjosh** [noir-lang/noir#3321] [Ideas]
### Problem

It is easy to make the Noir LSP crash. A missing curly brace will do it and it requires a window refresh to get it going again.

Will the nargo formatter help with this? Anything else we can/should do to improve the devex here?

### Happy Case

Noir LSP doesn't crash, or auto formats

### Would you like to submit a PR for this Issue?

No

**Comment by Savio-Sou** [comment]
How has your LSP been crashing?

Tried clicking the "Run Test" codelens with the missing } code snippet below. It prompted an error for sure, but pressing any arbitrary key dismissed the terminal and I'm good to go again:

<img width="1024" alt="Screenshot 2023-10-27 at 17 12 39" src="https://github.com/noir-lang/noir/assets/72797635/fe53aa1a-d145-4bb8-b834-2bf0b8feb8fe">

Was that what you experienced, or was it something else?

**Reply by critesjosh** [reply]
@colinnielsen does this match the behavior you mentioned on Friday?

I'll keep an eye out for the next time the LSP crashes for me and report the details here.

**Reply by colinnielsen** [reply]
Yes. Missing a curly brace causes consistent crashing of the in-IDE lsp

**Reply by Savio-Sou** [reply]
Thanks Colin 🙌
Were you able to close the terminal and simply proceed with using other LSP features without e.g. restarting VS Code?

**Reply by colinnielsen** [reply]
Unfortunately no, I had to manually find the error and restart vscode

**Reply by colinnielsen** [reply]
It was a painpoint in my workflow

**Comment by TomAFrench** [comment]
Discussions should not be used for bug reports. This should be made as an issue with a reproducible example.

**Comment by jfecher** [comment]
@critesjosh or @colinnielsen can you re-open this as an issue ideally with a reproducible example please? Sorry to cause some friction, but this helps us organize work a bit better!
